qiskit/algorithms/evolvers/variational/variational_principles/real_time_dependent_principle.py

```diff
@@ -0,0 +1,89 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class for a Real Time Dependent Variational Principle."""
+from typing import Union, Dict, List, Optional
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import Y, StateFn, CircuitQFI, OperatorBase, CircuitSampler
+from qiskit.opflow.gradients.circuit_gradients import LinComb
+from qiskit.opflow.gradients.circuit_qfis import LinCombFull
+from qiskit.utils import QuantumInstance
+from .real_variational_principle import (
+    RealVariationalPrinciple,
+)
+
+
+class RealTimeDependentPrinciple(RealVariationalPrinciple):
+    """Class for a Real Time Dependent Variational Principle. It works by evaluating the Lagrangian
+    corresponding the given system at a parametrized trial state and applying the Euler-Lagrange
+    equation. The principle leads to a system of linear equations handled by the
+    `~qiskit.algorithms.evolvers.variational.solvers.VarQTELinearSolver` class. The real variant
+    means that we consider real time dynamics.
+    """
+
+    def __init__(self, qfi_method: Union[str, CircuitQFI] = "lin_comb_full") -> None:
+        """
+        Args:
+            qfi_method: The method used to compute the QFI. Can be either
+                ``'lin_comb_full'`` or ``'overlap_block_diag'`` or ``'overlap_diag'`` or
+                ``CircuitQFI``.
+        """
+        if qfi_method == "lin_comb_full" or isinstance(qfi_method, LinCombFull):
+            qfi_method = LinCombFull(aux_meas_op=-Y)
```

Comment:
Good point. It might be that in this version we need to narrow the scope to ``LinCombFull`` only. @Zoufalc, did you consider other methods when adding measurements in other bases?
This is the relevant PR and it only addresses ``LinCombFull``.
https://github.com/Qiskit/qiskit-terra/pull/7632

Reply:
I narrowed it down for this var principle.